Fhi.HelseId.Integration.Tests/HelseId.Web/Tests/AuthenticationTests.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Net;
+using System.Web;
+using Fhi.HelseId.Integration.Tests.TestFramework;
+using Fhi.HelseId.Web;
+using Fhi.HelseId.Web.ExtensionMethods;
+using Fhi.TestFramework.Extensions;
+
+namespace Fhi.HelseId.Integration.Tests.HelseId.Web.Tests
+{
+    /// <summary>
+    /// The purpose of these tests is to test the buildt in OIDC authentication scheme in AddHelseIdWebAuthentication
+    /// </summary>
+    public class AuthenticationTests
+    {
+        [Test]
+        public async Task DefaultHelseIdConfiguration_NoAuthCookieOnApiCall_Return401WithRedirectToIdentityProvider()
+        {
+            var config = HelseIdWebKonfigurasjonBuilder.Create.AddDefaultValues()
+                .CreateConfigurationRoot();
+            var appFactory = new TestWebApplicationFactory(config, services =>
+            {
+                services.AddHelseIdWebAuthentication(config)
+               .Build();
+            });
+
+
+            var client = appFactory.CreateClient();
+            var response = await client.GetAsync("/api/test");
+
+            Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.Unauthorized));
+            var queryParams = HttpUtility.ParseQueryString(response.Headers.Location is not null ? response.Headers.Location.Query : string.Empty );
```

Comment:
Avoid logic in tests, if response.Headers.Location is null it should crash/fail ?

Reply:
Will fix